net.go

```diff
@@ -78,3 +81,85 @@ func Listen(network, address string) (net.Listener, error) {
 	// I asked about it here: https://groups.google.com/g/golang-nuts/c/Q1I7Viz9AJc
 	return defaultListenConfig().Listen(context.Background(), network, address)
 }
+
+type Listener struct {
+	net.Listener
+	log              log.Logger
+	tlsCfg           *tls.Config
+	handshakeTimeout time.Duration
+
+	// OnAccept is called when a new connection is successfully accepted.
+	OnAccept func(net.Conn)
```

Comment:
An interface with the callback would be better if we are planning to have more of them.